### PR TITLE
Userlifecycle

### PR DESCRIPTION
**Pull request by jasperRob**
This pull request creates a UserLifecycle workflow intended to handle generic keycloak user actions. Additionally, there is a fix put in QwandaUtils mandatory check.

### CODE REVIEW

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/route/Events.java [hunk 1]
```diff
@@ -173,18 +173,13 @@ public void route(QEventMessage msg) {
 			String prefix = CacheUtils.getObject(userToken.getProductCode(), Prefix.DEF_ + code + ":PREFIX", String.class);
 
 			JsonObject json = Json.createObjectBuilder()
-			.add("definitionCode", "DEF_".concat(code))
-			.add("sourceCode", userToken.getUserCode())
-			.build();
+				.add("definitionCode", "DEF_".concat(code))
```
Comment (Expressionless) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/route/Events.java]: Should use Prefix.DEF_ here

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/utils/KogitoScopeInit.java [hunk 2]
```diff
@@ -65,11 +65,6 @@ public void init(TaskExchange data) {
 		try {
 			userToken.init(userTokenStr);
 
-			// userToken.setProductCode(productCode);
-
-			// beUtils = Arc.container().instance(BaseEntityUtils.class);
-			beUtils = new BaseEntityUtils(serviceToken, userToken);
```
Comment (Expressionless) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/utils/KogitoScopeInit.java:71]: Is this really unable to be retrieved CDI? CommonUtils.getArcInstance is a safe replacement for Arc.container().instance

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/route/Events.java (hunk 1 above)
Comment (jasperRob) [reply kogitoq/gadaq/src/main/java/life/genny/gadaq/route/Events.java]: Good point

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java [hunk 3]
```diff
@@ -496,36 +496,43 @@ public static boolean mandatoryFieldsAreAnswered(Map<String, Ask> map, BaseEntit
...
-			if ((mandatory && !readonly))
+			Boolean answered = false;
+			if ((mandatory && !readonly)) {
 				answered = acceptableAnswer(value);
 
-			if(!answered) {
-				break;
+				// not complete if any mandatories are not answered
+				if (!answered) {
+					complete = false;
```
Comment (Expressionless) [inline qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java:505]: Do we need to keep checking which answers aren't complete if we have reached an incomplete field?

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java [hunk 4]
```diff
@@ -944,14 +949,13 @@ public Boolean isDuplicate(Definition definition, Answer answer, BaseEntity... t
 					// get the first value in array of target
 					for (BaseEntity target : targets) {
 
-						log.info("TARGET = " + target.getCode() + ", EMAIL = " + target.getValueAsString(Attribute.PRI_EMAIL));
-
 						if (target.containsEntityAttribute(code)) {
 							value = target.getValueAsString(code);
-							if (value.isEmpty())
+							if (value == null || value.isEmpty()) {
```
Comment (Expressionless) [inline qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java]: This can be reduced to StringUtils.isBlank(value)

**Review by Expressionless — CHANGES_REQUESTED** [review]
Left comments on small changes

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java (hunk 3 above)
Comment (jasperRob) [reply qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java:505]: We will come back to this.

**Review by varunshastry — APPROVED** [review]

**Review by Expressionless — APPROVED** [review]
lgtm

**Pull request merged**